cmd/kubeadm/app/cmd/phases/util.go

```diff
@@ -65,9 +65,11 @@ func runCmdPhase(cmdFunc func(outDir string, cfg *kubeadmapi.InitConfiguration)
 	}
 }
 
-// SetKubernetesVersion gets current Kubeadm version and sets it as Kubernetes version of master configuration if not set.
+// SetKubernetesVersion gets the current Kubeadm version and sets it as KubeadmVersion in the config,
+// unless it's already set to a value different from the default.
 func SetKubernetesVersion(client clientset.Interface, cfg *kubeadmapiv1beta1.InitConfiguration) error {
-	if cfg.KubernetesVersion != "" {
+
+	if cfg.KubernetesVersion != kubeadmapiv1beta1.DefaultKubernetesVersion {
```

Comment:
IIUC, the func will return from here if `cfg.KubernetesVersion == ""`. This is not what we expect, right?